### PR TITLE
feat(ngRoute):change pathRegExp to respect optional wildcard route params

### PR DESCRIPTION
**Pull request by murphomatic**
change pathRegExp to accept the two-character option of '*?' on route params to allow them to be wildcard and optional at the same time.  Motivation for change inspired by the need to accept an auth token on the URL that contained slashes (need for wildcard param) but this token on the URL was not required (need for optional param).  Previous behavior only saw the param as wildcard and not optional, which resulted in the need for a trailing slash on the url (e.g. /login/ vs. /login)

new feature - does not address any open issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,13 +32,13 @@
...
     "glob": "^6.0.1",
-    "grunt": "~0.4.2",
+    "grunt": "https://registry.npmjs.org/grunt/-/grunt-0.4.5.tgz",
     "grunt-bump": "~0.0.13",
     "grunt-contrib-clean": "~0.6.0",
     "grunt-contrib-compress": "~0.12.0",
     "grunt-contrib-connect": "~0.8.0",
     "grunt-contrib-copy": "~0.6.0",
-    "grunt-contrib-jshint": "~0.10.0",
+    "grunt-contrib-jshint": "https://registry.npmjs.org/grunt-contrib-jshint/-/grunt-contrib-jshint-0.10.0.tgz",
```
Comment (gkalpak) [inline package.json]: These changes should be removed.

File: test/ngRoute/routeSpec.js [hunk 2]
```diff
@@ -974,6 +974,32 @@ describe('$route', function() {
...
 
 
+    it('should properly process route params which are both eager and optional', function() {
+      module(function($routeProvider) {
+        $routeProvider.when('/foo/:param1*?/:param2', {templateUrl: 'foo.html'});
+      });
+
+      inject(function($location, $rootScope, $route) {
+        $location.path('/foo/bar1/bar2/bar3/baz');
+	$rootScope.$digest();
```
Comment (gkalpak) [inline test/ngRoute/routeSpec.js]: Indentation is off.

**Comment by gkalpak** [comment]
LGTM with the comments addressed :+1:

**Comment by murphomatic** [comment]
apologies for the miss, everything should be set now.

**Comment by murphomatic** [comment]
@gkalpak Let me know if there is any other work you see that needs to be done (I believe I've satisfied the issues you've commented on).  Thank you!

**Comment by gkalpak** [comment]
Merged ! Thx @murphomatic :+1: 
Backported to `v1.5.x` (c0797c6) and `v1.4.x` (6a4403a).

**Comment by murphomatic** [comment]
Awesome! Thanks @gkalpak!

**Pull request closed without merging**